includes/models/model.llms.student.php

```diff
@@ -1856,41 +1889,60 @@ private function update_completion_status( $status, $object_id, $object_type, $t
+			 * either "complete" or "incomplete"
 			 *
-			 * @action  llms_mark_complete
-			 * @action  llms_mark_incomplete
+			 * @since Unknown
+			 *
+			 * @param int    $student_id  WP_User ID of the student.
+			 * @param int    $object_id   WP_Post ID of the object.
+			 * @param string $object_type The type of object. A lesson, section, course, or course_track.
+			 * @param string $trigger     String describing the reason for the status change.
 			 */
-			do_action( 'llms_mark_' . $status, $this->get_id(), $object_id, $object_type, $trigger );
+			do_action( "llms_mark_{$status}", $student_id, $object_id, $object_type, $trigger );
 
 			/**
-			 * Specific hook
-			 * Also backwards compatible
+			 * Hook that fires when a student's completion status is updated for a specific object type.
+			 *
+			 * The dynamic portion of this hook, `$object_type` refers to the WP_Post post_type of the object
+			 * which the student's completion status is being updated for.
 			 *
-			 * @action  lifterlms_{$object_type}_completed
-			 * @action  lifterlms_{$object_type}_incompleted
+			 * The dynamic portion of this hook, `$status`, refers to the new completion status of the object,
+			 * either "complete" or "incomplete"
+			 *
+			 * @since Unknown
+			 *
+			 * @param int $student_id WP_User ID of the student.
+			 * @param int $object_id  WP_Post ID of the object.
 			 */
-			do_action( 'lifterlms_' . $object_type . '_' . $status . 'd', $this->get_id(), $object_id );
-
-			// cascade up for parents
-			if ( $parent_ids && $parent_type ) {
-
-				foreach ( $parent_ids as $pid ) {
-
-					$this->update_completion_status( $status, $pid, $parent_type, $trigger );
+			do_action( "lifterlms_{$object_type}_{$status}d", $student_id, $object_id );
 
+			// Cascade up for parents.
+			if ( $parent_data['ids'] && $parent_data['type'] ) {
+				foreach ( $parent_data['ids'] as $pid ) {
+					$this->update_completion_status( $status, $pid, $parent_data['type'], $trigger );
 				}
 			}
 
 			/**
-			 * Generic after hook
+			 * Hook that fires after a student's completion status for an object and it's parents have
+			 * been updated.
 			 *
-			 * @action  after_llms_mark_complete
-			 * @action  after_llms_mark_incomplete
+			 * The dynamic portion of this hook, `$status`, refers to the new completion status of the object,
+			 * either "complete" or "incomplete"
+			 *
+			 * @since Unknown
+			 *
+			 * @param int    $student_id  WP_User ID of the student.
+			 * @param int    $object_id   WP_Post ID of the object.
+			 * @param string $object_type The type of object. A lesson, section, course, or course_track.
+			 * @param string $trigger     String describing the reason for the status change.
 			 */
-			do_action( 'after_llms_mark_' . $status, $this->get_id(), $object_id, $object_type, $trigger );
+			do_action( "after_llms_mark_{$status}", $student_id, $object_id, $object_type, $trigger );
 
-		}// End if().
+		}
```

Comment:
Do you think adding to all the action hooks also `$this` instance maybe would be more helpful for 3rd parties?
I mean yeah they can always create a new instance of an LLMS_Student starting from the `$student_id` if needed but you know...

Reply:
since historically the actions didn't reference objects I don't know if it's essential to add another object to a set of triggers that already have a large number of arguments. Yes, it could be helpful but it's honestly not essential. Anyone hooking into the action has a at most 3 lines of code to instantiate (and check against a falsy).

I think we're going to leave it as is.